### PR TITLE
Pin Rust to 1.81.0

### PR DESCRIPTION
**Pull request by unexge**
- Create `rust-toolchain.toml`
- Use `actions-rust-lang/setup-rust-toolchain` GitHub action as it supports `rust-toolchain.toml`
---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Comment by passaro** [comment]
@unexge, is the `rust-toolchain` action actually required? My understanding is that cargo will respect `rust-toolchain.toml` automatically.

**Comment by unexge** [comment]
@passaro, not sure. It's [a popular request](https://github.com/dtolnay/rust-toolchain/issues?q=rust-toolchain+file) for [`dtolnay/rust-toolchain`](https://github.com/dtolnay/rust-toolchain) to support rust-toolchain file, so I assumed it won't work. But let me give it a try.

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -51,8 +52,10 @@ jobs:
         fuseVersion: 2
         libunwind: true
         fio: true
-    - name: Set up stable Rust
-      uses: dtolnay/rust-toolchain@stable
+    - name: Set up Rust toolchain
+      uses: dtolnay/rust-toolchain@master
```
Comment (passaro) [inline .github/workflows/bench.yml]: My suggestion was to remove this step entirely, not to revert to `stable`

Comment (passaro) [reply .github/workflows/bench.yml]: Alternative: we could look at some other actions, e.g. https://github.com/marketplace/actions/rust-toolchain-toml-install

Comment (unexge) [reply .github/workflows/bench.yml]: Agh, I misunderstood. I think `dtolnay/rust-toolchain` installs `rustup`/`cargo`. We might look installing it in `.github/actions/install-dependencies` step, not sure if there are any downsides.

Comment (dannycjones) [reply .github/workflows/bench.yml]: I was wondering about jumping to a new action, I saw this one that looks well maintained: https://github.com/actions-rust-lang/setup-rust-toolchain

Comment (unexge) [reply .github/workflows/bench.yml]: Updated to use `actions-rust-lang/setup-rust-toolchain`. I had to override `rustflags` option though, it's `-D warnings` by default which turns all warnings into compile errors, and `vendor/fuser` has lots of warnings. It'd be nice to get rid of those warnings eventually, but not turning this on should be fine for now.

This new action also surfaces warnings on PRs, e.g., scroll to the bottom on https://github.com/awslabs/mountpoint-s3/pull/1073/files. We can turn this of it we find it noisy, but eventually I'd like to address these warnings as well.

**Review by passaro — APPROVED** [review]

**Pull request merged**